src/vm_pages.rs

```diff
@@ -962,6 +962,10 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         page_addr: GuestPageAddr,
         num_pages: u64,
     ) -> Result<LockedPageList<Page<ConvertedDirty>>> {
+        if num_pages == 0 {
+            return Err(Error::EmptyPageRange);
+        }
```

Comment:
Should this check be in do_map_pages() too?

Reply:
Good point...added.